### ISSUE TITLE
The given key was not present in the dictionary.

### ISSUE THREAD

**Issue by coolgal1980** [valdisiljuconoks/LocalizationProvider#180]
Keeping getting the error when I try to upgrade Localization Provider Nuget Packages
The given key was not present in the dictionary.
   at System.Collections.Concurrent.ConcurrentDictionary`2.get_Item(TKey key)
   at DbLocalizationProvider.TypeFactory.GetHandler(Type queryType)
   at DbLocalizationProvider.TypeFactory.GetCommandHandler[TWrapper,TCommand](TCommand request, Type wrapperType)
   at DbLocalizationProvider.ICommandExtensions.CanBeExecuted(ICommand command)
   at DbLocalizationProvider.Sync.Synchronizer.UpdateStorageSchema()
   at DbLocalizationProvider.Sync.Synchronizer.DiscoverReadMerge()
   at DbLocalizationProvider.IAppBuilderExtensions.UseDbLocalizationProvider(IAppBuilder builder, Action`1 setup)
   at DbLocalizationProvider.EPiServer.DbLocalizationProviderInitializationModule.SetupLocalizationProvider(Object sender, EventArgs eventArgs)

**Comment by valdisiljuconoks** [comment]
Hi,

Can you paste your init module where you setup provider?

**Comment by coolgal1980** [comment]
I am not able to see it in the init module. What should I look for in the code? But this error started with the recent upgrade of the LocalizationProvider. I never had a problem with it before. Attaching some of the
[Business.zip](https://github.com/valdisiljuconoks/LocalizationProvider/files/4512814/Business.zip)
 code here for your reference.

**Comment by valdisiljuconoks** [comment]
yes, this error might show up in recent v6 version. this major version introduced separate SQL server package that you need to install and configure provider to use that storage by passing connection string to it. Please referrer to docs to get started.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by coolgal1980** [comment]
I read docs and installed LocalizationProvider.Storage.SqlServer.6.0.2 along which it needed to install the following also:
Microsoft.Data.SqlClient.SNI.1.1.0
Microsoft.Identity.Client.3.0.8
Microsoft.IdentityModel.Logging.5.5.0
Microsoft.IdentityModel.Tokens.5.5.0
Microsoft.IdentityModel.JsonWebTokens.5.5.0
Microsoft.IdentityModel.Protocols.5.5.0
System.Data.Common.4.3.0
Microsoft.IdentityModel.Protocols.OpenIdConnect.5.5.0
Microsoft.Data.SqlClient.1.1.0
LocalizationProvider.Storage.SqlServer.6.0.2

Can you please help with what I need to configure cuz after installing the above, I still get the error:
here is the startup.cs, the instructions in the docs are not clear.
[Startup.zip](https://github.com/valdisiljuconoks/LocalizationProvider/files/4513988/Startup.zip)

**Comment by valdisiljuconoks** [comment]
Yes, Microsoft.Data.SqlClient is pulling in some additional dependencies.

Agree that docs are bit distributed around different runtimes and it's sometimes hard to follow. Will see what can I do about this.

Anyways, I see that you have Episerver in your exception stack, then I suppose you are using provider in Episerver site.

For now, you can refer to sample AlloyTech site for how to setup provider: https://github.com/valdisiljuconoks/localization-provider-epi/blob/master/tests/DbLocalizationProvider.EPiServer.Sample/App_Start/DbLocalizationProviderConnectionSetupModule.cs#L30

**Comment by coolgal1980** [comment]
That helped, not getting the error anymore!! Thank you, had been struggling this for a while!
Now I am wondering if I really needed to install this package:
LocalizationProvider.Storage.SqlServer.6.0.2 and some of the others that got installed as dependencies with it??? 
Meaning if all that was missing was this configuration step of DBlocalizaionProvideConnectionSetupModule??

**Comment by valdisiljuconoks** [comment]
`.UseSqlServer()` extension method is coming from `LocalizationProvider.Storage.SqlServer` package - so you  basically need this. unless you are using provider with no underlying persistent storage.
